### PR TITLE
feat: Kafka production preset — CPU-sized QueueCount + NumPartitions (#684)

### PR DESCRIPTION
**Pull request by nightBaker**
## Summary

Closes #684. Adds opt-in `WithProductionDefaults(name)` chained extension for CPU-sized Kafka stream queue/partition tuning.

- **New file** `KafkaProductionPresetExtensions.cs` — internal `KafkaProductionPreset` math helper + public `WithProductionDefaults` extension that applies `QueueCount = max(8, ProcessorCount)` and `NumPartitions = max(1, ProcessorCount)` via `AddOptions<KafkaStreamingOptions>(name).Configure<ILoggerFactory>(...)`.
- **Startup INFO log** (EventId 11000) emitted on first options resolve — absence of the log is the observable signal for a provider-name mismatch (the silent no-op footgun from round-2 review).
- **7 + 2 tests** — pure math tests on `KafkaProductionPreset` helper (no host required, per round-2 recommendation) + 2 regression guards for unchanged defaults without preset.
- **`InternalsVisibleTo`** for `Fleans.Streaming.Kafka.Tests` added to access the internal helper.
- **Docs** — `docs/conventions/streaming.md` "Kafka production preset" section (usage, name-matching requirement, homogeneity requirement, forward-only NumPartitions caveat); EventId 11000–11099 range in `observability-eventids.md`.

## Key decisions

- Math extracted to `KafkaProductionPreset` (internal) so tests avoid building a full silo host — addresses round-2 Important #1.
- `Configure<ILoggerFactory>` pattern (not `Configure`) to capture logger for the startup observable — addresses round-2 Important #2 option (b).
- No startup WARN/ERROR log for homogeneity — deferred to #699 (cross-provider sanity probe), per round-2 question answer; docs document the risk instead.

## Test plan

- `dotnet test Fleans.Streaming.Kafka.Tests/ --filter KafkaProduction` — 7 preset tests pass.
- `dotnet build` — 0 errors.
- Manual: deploy with `WithProductionDefaults("kafka")` and verify EventId 11000 appears in startup logs; deploy with mismatched name and verify 11000 is absent and `QueueCount` stays at 8.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by nightBaker — COMMENTED** [review]
-

**Review by nightBaker — COMMENTED** [review]
## Code Review — feat: Kafka production preset — CPU-sized QueueCount + NumPartitions (#684)

### ✅ CLAUDE.md compliance
- `[LoggerMessage]` on `private static partial void` in a `static partial class` — valid static-class form of the source-generator pattern. ✓
- EventId 11000 allocated and documented in `observability-eventids.md`. ✓
- No `ILogger.Log*()` extension calls — the generated method is called directly. ✓
- Docs updated in same PR (streaming.md + observability-eventids.md). ✓
- No changes to grain state, no persistence concerns. ✓

### ✅ Design correctness
- Override ordering holds: `AddKafkaStreams` uses `b.Configure<KafkaStreamingOptions>` (plain Configure, not PostConfigure); `WithProductionDefaults` registers a later named Configure via `AddOptions<T>().Configure<ILoggerFactory>()` — runs after config bind and wins. Confirmed by round-2 analysis review. ✓
- Math extracted to `internal static KafkaProductionPreset` — pure functions, testable without a host. ✓
- `InternalsVisibleTo("Fleans.Streaming.Kafka.Tests")` correctly added alongside the existing `Fleans.Application.Tests`. ✓
- `ILoggerFactory` is always registered in the Orleans DI container; `Configure<ILoggerFactory>` is safe to use here. ✓
- The startup INFO log (EventId 11000) gives operators an observable signal to detect the silent-name-mismatch footgun. ✓

### ✅ Tests
9 tests (7 new + 2 existing regression guards): boundary-value coverage on `QueueCount` (floor=8, at-8, above-8) and `NumPartitions` (at-1, above-1), plus both regression guards confirming defaults are unchanged without preset. No host needed — direct invocation of `KafkaProductionPreset` helpers. ✓

### 🟢 Minor

**`Environment.ProcessorCount` read twice.** `queueCount`/`numPartitions` are computed at `WithProductionDefaults` call time (registration), but the log message re-reads `Environment.ProcessorCount` at lambda execution time (options resolution):
```csharp
LogProductionPresetApplied(..., name, queueCount, numPartitions, Environment.ProcessorCount);
//                                                                ↑ re-read here, not closure-captured
```
In practice `Environment.ProcessorCount` is immutable at runtime, so this is cosmetic. A follow-up could capture `processorCount` alongside `queueCount`/`numPartitions` for strict coherence. Not blocking.

**Log fires on each options resolution.** `.Configure<ILoggerFactory>` runs whenever the named-options value is freshly computed. With static configuration this is once per silo start — acceptable noise level.

No blocking issues. OK to promote.

File: src/Fleans/Fleans.Streaming.Kafka/Fleans.Streaming.Kafka.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
   <ItemGroup>
     <InternalsVisibleTo Include="Fleans.Application.Tests" />
+    <InternalsVisibleTo Include="Fleans.Streaming.Kafka.Tests" />
```
Comment (nightBaker) [inline src/Fleans/Fleans.Streaming.Kafka/Fleans.Streaming.Kafka.csproj:13]: why it includes tests ?
